packages/design-system-react-native/src/components/Icons/Icon/scripts/generate-assets.scripts.ts

```diff
@@ -0,0 +1,141 @@
+    const fileContent = await fs.promises.readFile(filePath, {
+      encoding: 'utf-8',
+    });
+    const formattedFileContent = fileContent.replace(/black/gu, 'currentColor');
+    await fs.promises.writeFile(filePath, formattedFileContent);
+  }
+
+  await fs.promises.writeFile(assetsModulePath, '');
+
+  await fs.promises.appendFile(
+    assetsModulePath,
+    `/* eslint-disable import/prefer-default-export */`,
+  );
+
+  await fs.promises.appendFile(
+    assetsModulePath,
+    `\n// /////////////////////////////////////////////////////\n// This is a generated file\n// DO NOT EDIT - Use generate-assets.js\n// /////////////////////////////////////////////////////`,
+  );
+
+  // Append the SVG imports first
+  for (const fileName of assetFileList) {
+    const iconName = getIconNameInTitleCase(fileName);
+    await fs.promises.appendFile(
+      assetsModulePath,
+      `\nimport ${iconName}SVG from './assets/${fileName}';`,
+    );
+  }
+
+  // Separate IconName (runtime) and AssetByIconName (type-only) imports
+  await fs.promises.appendFile(
+    assetsModulePath,
+    `\nimport { IconName } from './Icon.types';`,
+  );
+  await fs.promises.appendFile(
+    assetsModulePath,
+    `\nimport type { AssetByIconName } from './Icon.types';`,
+  );
+
+  await fs.promises.appendFile(
+    assetsModulePath,
+    `\n\n/**\n * Asset stored by icon name\n */`,
+  );
+
+  await fs.promises.appendFile(
+    assetsModulePath,
+    `\nexport const assetByIconName: AssetByIconName = {`,
+  );
+
+  for (const fileName of assetFileList) {
+    const iconName = getIconNameInTitleCase(fileName);
+    await fs.promises.appendFile(
+      assetsModulePath,
+      `\n  [IconName.${iconName}]: ${iconName}SVG,`,
+    );
+  }
+
+  await fs.promises.appendFile(assetsModulePath, '\n};\n');
+
+  const typesFileContent = await fs.promises.readFile(typesFilePath, {
+    encoding: 'utf8',
+  });
+  const indexToRemove = typesFileContent.indexOf(TYPES_CONTENT_TO_DETECT);
+  const baseTypesFileContent = typesFileContent.substring(0, indexToRemove);
+
+  let typesContentToWrite = `${
+    baseTypesFileContent + TYPES_CONTENT_TO_DETECT
+  }\n// /////////////////////////////////////////////////////`;
+
+  typesContentToWrite += '\n\n/**\n * Icon names\n */\nexport enum IconName {';
+
+  for (const fileName of assetFileList) {
+    const iconName = path
+      .basename(fileName, ASSET_EXT)
+      .split('-')
+      .map(
+        (section) =>
+          `${section[0].toUpperCase()}${section.substring(1, section.length)}`,
+      )
+      .join('');
+    typesContentToWrite += `\n  ${iconName} = '${iconName}',`;
+  }
+
+  typesContentToWrite += '\n}\n';
+
+  await fs.promises.writeFile(typesFilePath, typesContentToWrite);
+
+  console.log(`✅ Finished assets file! 🎉🎉`);
+}
+
+if (require.main === module) {
+  main().catch((error) => {
+    console.error(error);
+    throw error; // Throw instead of process.exit(1)
+  });
+}
```

Comment:
Should this live in a scripts folder in src? `src/scripts` and what was your reason for not writing tests for this file? I was able to get the testing coverage up to `93.75%` with some help from cursor


<img width="911" alt="Screenshot 2024-12-12 at 2 40 28 PM" src="https://github.com/user-attachments/assets/17fe59f4-c2e3-4ee2-919e-da2e4720641b" />

Reply:
There is a lot of file manipulation with the script, including amending the existing types and assets file. I did try to create a test for this but havent been successful with it. It can live in src/scripts if that's what you prefer

Reply:
Maybe we can create an issue to add tests for this script in future if you aren't able to get them working in this PR?

Reply:
https://github.com/MetaMask/metamask-design-system/issues/296